readme.md

```diff
@@ -42,7 +42,18 @@ That way you are in control of the order in which things are executed.
 
 
 ## Testing
+There should be unit tests for your code. A unit test should be independent and do an assert for each unit. The tests for business logic should be independent so it has no dependencies on database or external services. An exception is a repository test -- there you want to test a database transaction.
 
+The unit test should be clean and treated as part of the codebase. They are part of your deployment process.
+
+The purpose of a unit test -- it should give quick feedback to check if the production code does what we expect it to do. The unit test is not responsible for testing the infrastructure and application chain. We don’t have to test the framework. The unit test does not know in what context the code will run, so therefore it should be independent and mocking should always be used when necessary.
+
+For example, when testing functionality that is called from a trigger, you want to unit-test the scoped pieces of code that will be executed separately. For example, a unit test will not test the whole process of a record going through a trigger, or a process from UI to database; a unit test is not the place to do that.
+
+A unit test will also help with designing your code. With TDD you will write unit tests and scenarios. After that, you will improve the code, so that it will make only these tests pass.
```

Comment:
"so that it will make only these tests pass"? what do you mean with that?

Reply:
You write code only with the purpose to make the tests green. So no unnecessary code.